basyx.aasenvironment/basyx.aasenvironment-core/src/main/java/org/eclipse/digitaltwin/basyx/aasenvironment/base/DefaultAASEnvironmentSerialization.java

```diff
@@ -37,54 +39,101 @@
 	private SubmodelRepository submodelRepository;
+	private ConceptDescriptionRepository conceptDescriptionRepository;
 	private Serializer jsonSerializer = new JsonSerializer();
 	private Serializer xmlSerializer = new XmlSerializer();
 	private AASXSerializer aasxSerializer = new AASXSerializer();
-
-	public DefaultAASEnvironmentSerialization(AasRepository aasRepository, SubmodelRepository submodelRepository) {
+	private MetamodelCloneCreator cloneCreator = new MetamodelCloneCreator();
+	
+	public DefaultAASEnvironmentSerialization(AasRepository aasRepository, SubmodelRepository submodelRepository, ConceptDescriptionRepository conceptDescriptionRepository) {
 		this.aasRepository = aasRepository;
 		this.submodelRepository = submodelRepository;
+		this.conceptDescriptionRepository = conceptDescriptionRepository;
 	}
 
 	@Override
-	public String createJSONAASEnvironmentSerialization(@Valid List<String> aasIds, @Valid List<String> submodelIds) throws SerializationException {
-		Environment aasEnvironment = extractShellsAndSubmodels(aasIds, submodelIds);
+	public String createJSONAASEnvironmentSerialization(@Valid List<String> aasIds, @Valid List<String> submodelIds, @Valid boolean includeConceptDescriptions) throws SerializationException {
+		Environment aasEnvironment = extractShellsAndSubmodels(aasIds, submodelIds, includeConceptDescriptions);
 		return jsonSerializer.write(aasEnvironment);
 	}
 
 	@Override
-	public String createXMLAASEnvironmentSerialization(@Valid List<String> aasIds, @Valid List<String> submodelIds) throws SerializationException {
-		Environment aasEnvironment = extractShellsAndSubmodels(aasIds, submodelIds);
+	public String createXMLAASEnvironmentSerialization(@Valid List<String> aasIds, @Valid List<String> submodelIds, @Valid boolean includeConceptDescriptions) throws SerializationException {
+		Environment aasEnvironment = extractShellsAndSubmodels(aasIds, submodelIds, includeConceptDescriptions);
 
 		return xmlSerializer.write(aasEnvironment);
 	}
 
 	@Override
-	public byte[] createAASXAASEnvironmentSerialization(@Valid List<String> aasIds, @Valid List<String> submodelIds) throws SerializationException, IOException {
-		Environment aasEnvironment = extractShellsAndSubmodels(aasIds, submodelIds);
+	public byte[] createAASXAASEnvironmentSerialization(@Valid List<String> aasIds, @Valid List<String> submodelIds, @Valid boolean includeConceptDescriptions) throws SerializationException, IOException {
+		Environment aasEnvironment = extractShellsAndSubmodels(aasIds, submodelIds, includeConceptDescriptions);
 
 		ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
 		aasxSerializer.write(aasEnvironment, null, outputStream);
 		return outputStream.toByteArray();
 	}
 
-	private Environment extractShellsAndSubmodels(List<String> aasIds, List<String> submodelIds) {
+	private Environment extractShellsAndSubmodels(List<String> aasIds, List<String> submodelIds, boolean includeConceptDescriptions) {
```

Comment:
The function name does reflect the intent anymore since now conceptDescriptions may be included